src/backend/access/appendonly/aosegfiles.c

```diff
@@ -1501,6 +1507,7 @@ get_ao_distribution(PG_FUNCTION_ARGS)
 	 */
 	pfree(query_block);
 
+srf_done:
 	SPI_finish();
```

Comment:
I think the `srf_done` label could be moved higher, and it would also be worth adding checks like this:
```c
srf_done:
	if (query_block != NULL)
		pfree(query_block);

	if (connected)
		SPI_finish();
```